### PR TITLE
Enumerable Project - Pull Request

### PR DESCRIPTION
**Pull request by phfdonda**
Complete project.
Thank you for your review and any insight you can give me! This was a very hard assignment, and any feedback would be very welcome! Thank you!

### CODE REVIEW

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi @phfdonda     
#### Great Job so far, but there are a few more things required. 
### General
- [ ] Lets enable stickler 
- [ ] Pull Request should have a meaningful title
- [ ] Next project use a better branch name for example `enum_methods`
- [ ] `my_inject`:
- When a symbol is specified combines each element of the collection by applying the symbol as a named method
`p (5..10).my_inject(2, :*) - should return 302400`

- [ ] `my_none?`:
- When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
`p ['hello'].my_none?(5)` - should return `true`
- [ ] `my_any?`:
- when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
`p ['cat',2,32,].my_any?('cat')` should return `true`
- when no block or argument is given returns true if at least one of the collection is not false or nil
`p [nil, false, nil, false].my_any?` - should return `false`



- [ ] `my_all`  
- when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
`p [3,3,3,3,4].my_all?(3)` should return` false`



### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Comment by phfdonda** [comment]
I have made the changes requested, @kobiyoyo! Thank you for the excellent review!

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+      my_each { |i| return false unless i.match(arg) }
+    elsif arg.is_a? Module
+      my_each { |i| return false unless i.is_a?(arg) }
+    elsif !block_given? || arg.nil?
+      my_each { |i| return false unless i.nil? || i == false }
+    else
+      my_each { |i| return false if i.nil? || i == false }
+    end
+    true
+  end
```
Comment (stratospherique) [inline main.rb]: * [ ] `my_all?` is identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_all? when no block or argument is given returns true when none of the collection members are false or nil
  ```ruby
    array = [1,2,5,4,7]
    p array.my_all? == array.all? # false
  ```
* [ ] my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
  ```ruby
    array = [3,3,3,3,3]
    p array.my_all?(3) == array.all?(3) # false
  ```

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+      my_each { |i| return true if i.match(arg) }
+    elsif arg.is_a? Module
+      my_each { |i| return true if i.is_a?(arg) }
+    elsif !block_given? || arg.nil?
+      my_each { |i| return true unless i.nil? || i == false }
+    else
+      my_each { |i| return true if i.nil? || i == false }
+    end
+    false
+ end
```
Comment (stratospherique) [inline main.rb]: * [ ] `my_any?` is identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
  ```ruby
    array = %w[dog door rod blade]
    p array.my_any?('cat') == array.any?('cat') # false
  ```

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      my_each { |i| return false if i.match(arg) }
+    elsif arg.is_a? Module
+      my_each { |i| return false if i.is_a?(arg) }
+    elsif !block_given? || arg.nil?
+      my_each { |i| return false if i.nil? || i == false }
+    else
+      my_each { |i| return false unless i == false }
+    end
+    true
+  end
```
Comment (stratospherique) [inline main.rb]: * [ ] `my_none?` is identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_none? when no block or argument is given returns true only if none of the collection members is true
  ```ruby
    array = [nil, false, nil, false]
    p array.my_none? == array.none? # false
  ```

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great Progress @phfdonda :clap: However, some methods still needs further refactoring in order for us to approve your project. Check :arrow_down: 

- **[REQUIRED]** Check the code comments below, they should help you refactor your methods :gear: 

### That's it :relaxed: when finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Comment by phfdonda** [comment]
Thank you, @stratospherique! I made the changes, and I believe it's all working now! That was a good tip, comparing to the original method. Thank you!

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi @phfdonda     
#### Great Job so far, you are almost there .
### General

- [ ] `my_none?`:
- When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
`p ['hello',5].my_none?(5) `should return false




### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Comment by phfdonda** [comment]
Hi, @kobiyoyo. Thank you, silly mistake. Yeah, I think that's it now! Thank you for the review!

**Review by Torres-ssf — APPROVED** [review]
### Project Approved :tada:  :clap: 
Hi there @phfdonda 
Nice job implementing the requested changes.
All your methods are working properly now. :smile: 
**I'm approving this PR**
Good luck in the next project

**[TIP]**
- [ ] The .gitignore should be in the root of your project. And you should add both `.vscode` and `.idea`
(hidden on OS X). These two files are specific settings files(from vscode and IntelliJ’s) and should not be uploaded to GitHub.


Happy coding :computer: 
**Sergio Torres**

**Pull request merged**